cupy/random/distributions.py

```diff
@@ -202,6 +202,31 @@ def normal(loc=0.0, scale=1.0, size=None, dtype=float):
     return x
 
 
+def standard_exponential(size=None, dtype=float):
+    """Standard exponential distribution.
+
+    Returns an array of samples drawn from the standard exponential
+    distribution. Its probability density function is defined as
+
+      .. math::
+         f(x) = e^{-x},
+
+    Args:
+        size (int or tuple of ints): The shape of the array. If ``None``, a
+            zero-dimensional array is generated.
+        dtype: Data type specifier. Only :class:`numpy.float32` and
+            :class:`numpy.float64` types are allowed.
+
+    Returns:
+        cupy.ndarray: Samples drawn from the standard exponential distribution.
+
+    .. seealso:: :func:`numpy.random.standard_exponential`
+    """
+    rs = generator.get_random_state()
+    x = rs.uniform(0, 1, size, dtype)
+    return -cupy.log(1.0 - x)
```

Comment:
We cannot take the `1.0 - uniform` approach here, because `1.0 - (1.0 - x)` can be 0 for a nonzero `x` due to an round-off error.
```
>>> [float(cupy.random.uniform(size=10**7, dtype=cupy.float32).max()) for _ in range(10)]
[0.9999999403953552, 0.9999995827674866, 0.9999997615814209, 0.9999998807907104, 1.0, 0.9999998807907104, 0.9999999403953552, 0.9999998211860657, 0.9999998211860657, 0.9999999403953552]
```

Reply:
Umm...

> Samples are drawn from a uniform distribution over the half-open interval ``[low, high)``.

I feel that the implementation of `cupy.random.uniform` is not conforming to the specification.
But, I'll try to other approach...